### PR TITLE
LibJS: Fix parsing of mixed `CoalesceExpression` and `Logical{AND,OR}Expression`

### PR DESCRIPTION
**Pull request by linusg**

### CODE REVIEW

File: Userland/Libraries/LibJS/Parser.cpp [hunk 1]
```diff
@@ -2010,7 +2011,7 @@ NonnullRefPtr<Expression const> Parser::parse_expression(int min_precedence, Ass
             check_for_invalid_object_property(expression);
 
             Associativity new_associativity = operator_associativity(m_state.current_token.type());
-            auto result = parse_secondary_expression(move(expression), new_precedence, new_associativity, forbidden);
```
Comment (davidot) [inline Userland/Libraries/LibJS/Parser.cpp:2013]: Hmm not sure if this is correct for all cases, but if it doesn't break anything it's fine

File: Userland/Libraries/LibJS/Tests/syntax/coalesce-logic-expression-mixing.js [hunk 2]
```diff
@@ -20,6 +20,19 @@ test("mixing coalescing and logical operators with parens", () => {
...
+    expect("0 || 0 ? 0 : 0 ?? 0").toEval();
+    expect("0 ?? 0 ? 0 : 0 || 0").toEval();
+    expect("0 ? 0 || 0 : 0 ?? 0").toEval();
+    expect("0 ? 0 ?? 0 : 0 || 0").toEval();
+    expect("0 && 0 ? 0 ?? 0 : 0 || 0").toEval();
+    expect("0 ?? 0 ? 0 && 0 : 0 || 0").toEval();
+    expect("0 ?? 0 ? 0 || 0 : 0 && 0").toEval();
+    expect("0 || 0 ? 0 ?? 0 : 0 && 0").toEval();
+    expect("0 && 0 ? 0 || 0 : 0 ?? 0").toEval();
+    expect("0 || 0 ? 0 && 0 : 0 ?? 0").toEval();
```
Comment (davidot) [inline Userland/Libraries/LibJS/Tests/syntax/coalesce-logic-expression-mixing.js:33]: Were we only too strict?
I.e. there are no new cases we shouldn't eval?

**Review by davidot — APPROVED** [review]
That was a lot easier than I expected, this might even fix some other issues (or at least be one step towards fixing them).

Code LGTM, just some nitpicky questions which you can ignore

File: Userland/Libraries/LibJS/Parser.cpp (hunk 1 above)
Comment (linusg) [reply Userland/Libraries/LibJS/Parser.cpp:2013]: I ran test262 and it did nothing, good or bad :smile:

File: Userland/Libraries/LibJS/Tests/syntax/coalesce-logic-expression-mixing.js (hunk 2 above)
Comment (linusg) [reply Userland/Libraries/LibJS/Tests/syntax/coalesce-logic-expression-mixing.js:33]: "mixing coalescing and logical operators isn't allowed" above already covers various cases that are not allowed, and I couldn't think of anything else

**Pull request merged**